common/include/pcl/memory.h

```diff
@@ -80,6 +81,12 @@ namespace pcl
 template <typename T>
 using shared_ptr = boost::shared_ptr<T>;
 
+/** ADL doesn't work until C++20 for dynamic_pointer_cast since it requires an explicit Tparam */
+using boost::dynamic_pointer_cast;
```

Comment:
Let's use one format. Either switch `shared_ptr` to this or the other way around.

Reply:
I don't understand what you mean. Here I'm bringing both casts in, so as to force ADL like Morwenn suggested. Later in #3750 we will switch these from `boost::` to `std::` like `shared_ptr` and `weak_ptr`.

Reply:
L81-82 can be replaced by `using boost::shared_ptr;` and the code will remain consistent. Else you can define new symbols and that too will keep the code consistent